src/WebJobs.Script/OutOfProc/Http/DefaultHttpInvokerService.cs

```diff
@@ -3,23 +3,90 @@
 {
     public class DefaultHttpInvokerService : IHttpInvokerService
     {
         private readonly HttpClient _httpClient;
+        private readonly string _httpInvokerBaseUrl;
 
         public DefaultHttpInvokerService(HttpClient httpClient)
         {
             _httpClient = httpClient;
+            // TODO: user agent ok?
+            _httpClient.DefaultRequestHeaders.UserAgent.ParseAdd(HttpInvokerConstants.UserAgentHeaderValue);
+            _httpInvokerBaseUrl = "http://localhost:8090/";
         }
 
-        public Task GetInvocationResponse(ScriptInvocationContext scriptInvocationContext)
+        public async Task InvokeAsync(ScriptInvocationContext scriptInvocationContext)
         {
-            throw new NotImplementedException();
+            var requestUri = _httpInvokerBaseUrl + scriptInvocationContext.FunctionMetadata.Name;
+            HttpScriptInvocationContext httpScriptInvocationContext = new HttpScriptInvocationContext();
+
+            // populate metadata
+            foreach (var bindingDataPair in scriptInvocationContext.BindingData)
+            {
+                if (bindingDataPair.Value != null)
+                {
+                    if (bindingDataPair.Value is HttpRequest)
+                    {
+                        // no metadata for httpTrigger
+                        continue;
+                    }
+                    // TODO: how to handle byte[]? base64 encoded string?
+                    httpScriptInvocationContext.Metadata[bindingDataPair.Key] = JsonConvert.SerializeObject(bindingDataPair.Value);
+                }
+            }
+
+            // populate input bindings
+            foreach (var input in scriptInvocationContext.Inputs)
+            {
+                if (input.val is HttpRequest httpRequest)
+                {
+                    httpScriptInvocationContext.Data[input.name] = HttpRequestMessageConverters.ConvertHttpRequestToJObject(httpRequest);
+                    continue;
+                }
+                httpScriptInvocationContext.Data[input.name] = JsonConvert.SerializeObject(input.val);
+            }
+            HttpContent content = new ObjectContent<HttpScriptInvocationContext>(httpScriptInvocationContext, new JsonMediaTypeFormatter());
+            // Add standard headers
+            content.Headers.Add(HttpInvokerConstants.InvocationIdHeaderName, scriptInvocationContext.ExecutionContext.InvocationId.ToString());
+            content.Headers.Add(HttpInvokerConstants.HostVersionHeaderName, ScriptHost.Version);
+            try
+            {
+                HttpResponseMessage response = await _httpClient.PostAsync(requestUri, content);
+                HttpScriptInvocationResult invocationResult = await response.Content.ReadAsAsync<HttpScriptInvocationResult>();
+                ProcessLogsFromHttpResponse(scriptInvocationContext, invocationResult);
+                var result = new ScriptInvocationResult()
```

Comment:
For HTTP trigger scenarios, can we have something a bit simpler here where the custom worker does not have to return a special response type?

Reply:
Yes, we can figure out if this is a simple httpTrigger and forward the request to the invoker and use the response as function invocation response. Tracking issue https://github.com/Azure/azure-functions-host/issues/4992